CMakeLists.txt

```diff
@@ -68,42 +68,53 @@ if(${BUILD_PYTHON})
     endif()
 endif()
 
-if(MSVC)
-    # TODO: TEST this
-    set(COMPILE_FLAGS  /W4 /WX) 
-else()
+if (NOT MSVC)
     set(COMPILE_FLAGS  -Wall -Wextra  -Werror -MMD) #-Wno-deprecated
     set(RELEASE_FLAGS -O3) #-Wno-deprecated
 endif()
 
 if (RUN_MEMORY_TEST)
-    set(MEMORY_FLAGS -fsanitize=address -fno-omit-frame-pointer -O1)
+    if(NOT MSVC)
+        set(MEMORY_FLAGS -fsanitize=address -fno-omit-frame-pointer -O1)
+    endif()
 endif()
 
-set(DEBUG_FLAGS ${COMPILE_FLAGS} ${MEMORY_FLAGS} -g -DDEBUG_ON) 
-set(CXXTEST_FLAGS ${DEBUG_FLAGS}  -Wno-ignored-qualifiers)
+if (NOT MSVC)
+    set(DEBUG_FLAGS ${COMPILE_FLAGS} ${MEMORY_FLAGS} -g -DDEBUG_ON) 
+    set(CXXTEST_FLAGS ${DEBUG_FLAGS}  -Wno-ignored-qualifiers)
+endif()
 
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
     set(CXXTEST_FLAGS ${CXXTEST_FLAGS} -Wno-terminate)
 endif()
-set(RELEASE_FLAGS  ${COMPILE_FLAGS} -O3)
 
+if (NOT MSVC)
+    set(RELEASE_FLAGS  ${COMPILE_FLAGS} -O3)
+endif()
 
 
 # Boost
-if(${CMAKE_SIZEOF_VOID_P} EQUAL 4)
+if (${CMAKE_SIZEOF_VOID_P} EQUAL 4 AND NOT MSVC)
     set(BOOST_ROOT "${BOOST_DIR}/installed32")
     # set(BOOST_INCLUDES "${BOOST_DIR}/installed32/include")
 else()
     set(BOOST_ROOT "${BOOST_DIR}/installed")
     # set(BOOST_INCLUDES "${BOOST_DIR}/installed/include")
 endif()
+
+if (MSVC)
+    set(Boost_NAMESPACE libboost)
+endif()
 find_package(Boost COMPONENTS program_options)
 # Find boost
 if (NOT ${Boost_FOUND})
     # bash file that downloads and install boost 1_68_0, the name need to match
     # BOOST_DIR variable
-    execute_process(COMMAND ${TOOLS_DIR}/download_boost.sh)
+    if(MSVC)
```

Comment:
Actually I think we should check if it's WIN32 (aka a windows terminal) or not.
MSVC tells you whether we are using cl (visual studio compiler) or not

Reply:
I'm not sure if it will make too much difference. The download_boost.bat uses the msvc toolset to build the program_options library, so it's currently assumed that MSVC is going to be used from a windows machine.